engine/src/main/java/org/terasology/rendering/opengl/GLSLShader.java

```diff
@@ -322,85 +328,68 @@ private void compileShader(int type, Set<ShaderProgramFeature> features) {
             shader.append(shaderProgramBase.getVertexProgram());
         }
 
+        return shader.toString();
+    }
+
+    private void dumpCode(int type, Set<ShaderProgramFeature> features, String sourceCode) {
         String debugShaderType = "UNKNOWN";
         int featureHash = ShaderProgramFeature.getBitset(features);
         if (type == GL20.GL_FRAGMENT_SHADER) {
-            fragmentPrograms.put(featureHash, shaderId);
             debugShaderType = "FRAGMENT";
         } else if (type == GL20.GL_VERTEX_SHADER) {
-            vertexPrograms.put(featureHash, shaderId);
             debugShaderType = "VERTEX";
         }
 
-        Config config = CoreRegistry.get(Config.class);
-        if (config.getRendering().isDumpShaders()) {
-            // Dump all final shader sources to the log directory
-            final String strippedTitle = getURI().toString().replace(":", "-");
-
-            String fname = debugShaderType.toLowerCase() + "_" + strippedTitle + "_" + featureHash + ".glsl";
-            Path path = PathManager.getInstance().getShaderLogPath().resolve(fname);
-            try (BufferedWriter writer = Files.newBufferedWriter(path, TerasologyConstants.CHARSET)) {
-                writer.write(shader.toString());
-            } catch (IOException e) {
-                logger.error("Failed to dump shader source.");
-            }
+        // Dump all final shader sources to the log directory
+        final String strippedTitle = getURI().toString().replace(":", "-");
+
+        String fname = debugShaderType.toLowerCase() + "_" + strippedTitle + "_" + featureHash + ".glsl";
```

Comment:
Can you add a comment providing an example of what the filename would be like? I can sort of imagine it but it'd be good to have an example.